src/http/http_response.f90

```diff
@@ -36,6 +39,29 @@ subroutine append_header(this, key, value)
         end if
     
     end subroutine append_header
+
+    function get_header_value(this, find_key) result(retval)
+        class(response_type), intent(in) :: this
+        character(*), intent(in) :: find_key
+        character(:), allocatable :: retval
+        type(string_type) :: temp, h_key
+        type(header_type), allocatable :: header(:)
+        integer :: i
+        
+        header = this%header
+        temp = string_type(find_key)
+        temp = to_lower(temp)
+        
+        retval = ''
+
+        do i=1, size(header)
+            h_key = string_type(header(i)%key)
+            h_key = to_lower(h_key)
+            if(h_key == temp) then
```

Comment:
Since you do this in every iteration, you can just do:

```suggestion
            if (to_lower(string_type(header(i)%key) == string_to_match)) then
```